### ISSUE TITLE
jspm + typescript + bundle: ts sources not included in sourcemap

### ISSUE THREAD

**Issue by rfuerst87** [jspm/jspm-cli#1642]
Maybe I'm doing something terribly wrong, but I cannot get jspm to include typescript sources in the sourcemap using `bundle` or `bundle-sfx`. The setup is as follows:

```javascript
System.config({
  baseURL: "/",
  defaultJSExtensions: true,
  transpiler: "typescript",
  typescriptOptions: {
    "module": "system",
    "emitDecoratorMetadata": true,
    "sourceMap": true
  },
  paths: {
    "npm:*": "jspm_packages/npm/*",
    "github:*": "jspm_packages/github/*"
  },

  packages: {
    "app": {
      "defaultExtension": "ts"
    }
  },

  map: {
    ...
  }
});
```

For some reason `jspm bundle-sfx app/main.ts app.bundle.js` creates a sourcemap (`app.bundle.js.map`)  for all `jspm_packages` but not for typescript sources. Looking into the sourcemap ts sources cannot be found under `sources` nor under `sourcesContent`. Appending `--source-map-contents` to the bundle command didn't help either.

Am I missing something here? Is this even supported by jspm? jspm version used is 0.16.30.

**Comment by rfuerst87** [comment]
I played around with this issue a little today. Using [plugin-typescript](https://github.com/frankwallis/plugin-typescript) I managed to get source maps working but I ran immediately into another problem: [Source Maps are off](https://github.com/frankwallis/plugin-typescript/issues/110). An update to jspm 0.17.11 as suggested by @frankwallis didn't help.

Could this be a bug in system?

**Comment by born2net** [comment]
I am also seeing same issues with my breakpoint stopping at other places.. using latest jspm and angular2

**Comment by frankwallis** [comment]
I find that commenting out this line seems to help:

https://github.com/systemjs/systemjs/blob/bb588cee15403be37edb807b765fc69168014f36/dist/system.src.js#L1371

**Comment by develar** [comment]
@frankwallis See https://github.com/systemjs/systemjs/pull/1180

**Comment by guybedford** [comment]
@born2net is this resolved by https://github.com/systemjs/systemjs/pull/1180 then?

**Comment by rfuerst87** [comment]
@guybedford seems like sourcemaps are still off when using `jspm bundle | bundle-sfx`. Also the mapped lines seems to be kind of random. Could this be related to https://github.com/jspm/jspm-cli/issues/1104?

**Comment by born2net** [comment]
if using WebStorm you can follow this:
http://stackoverflow.com/questions/36228177/jspm-system-js-debugging-with-typescript

**Comment by frankwallis** [comment]
@guybedford - yes systemjs/systemjs#1180 resolves the issue with the source maps being 'off' when transpiling in the browser (https://github.com/frankwallis/plugin-typescript/issues/110)

**Comment by guybedford** [comment]
Great, thanks @frankwallis for the update.

**Comment by guybedford** [comment]
Released in jspm 0.17.0-beta.12.
